src/lib.rs

```diff
@@ -66,15 +68,32 @@ use {
   prometheus_metrics::serve_prometheus,
 };
 
+#[derive(Parser, Debug)]
+struct CmdArgs {
+  #[arg(long, default_value_t = false)]
+  /// if you start multiple lemmy server instances set this to true on all but one of them
+  disable_scheduled_tasks: bool,
+  /// set to false to disable the http server
+  #[arg(long, default_value_t = true, action=ArgAction::Set)]
+  http_server: bool,
+  /// set to false to disable the outgoing federation in this process
+  #[arg(long, default_value_t = true, action=ArgAction::Set)]
+  federate_activities: bool,
+  /// the index of this outgoing federation process. only useful if you want to split federation work into multiple servers.
+  /// the first process has number 1.
+  #[arg(long, default_value_t = 1)]
+  federate_process_index: i32,
+  /// how many outgoing federation processes you are starting in total. if set, make sure to set --activity-process-index differently for each.
```

Comment:
```suggestion
  /// How many outgoing federation processes you are starting in total.
  ///
  /// If set, make sure to set --federate_process_index differently for each.
```

Not sure but would it be a better idea to automatically detect somehow how many lemmy instances are already running? Or perhaps use UUIDs instead of integers starting from 1? Not sure if that makes sense as again, not sure about the use case of this argument.

Reply:
it needs to be sequential because it's a really simple load leveling where process i/n handles all rows from the db with `id % n == i`

Reply:
Okay that needs to be documented explicitly, as that sounds like it could be a big source of trouble if you don't set these correctly.

Reply:
Probably best to describe it on the documentation page "Horizontal scaling", and put a link to that.